src/data/bmg_plate.py

```diff
@@ -0,0 +1,101 @@
+import os
+import numpy as np
+import pandas as pd
+from collections import namedtuple
+
+
+class Plate:
+    """
+    Class representing a plate with values resulting from an HTS experiment
+    """
+
+    def __init__(self, barcode: str, plate_array: np.array) -> None:
+        """
+        :param filepath: path to the file consisting plate
+        """
+        self.barcode = barcode
+        self.plate_array = plate_array
+        self.control_statistics()
+        self.z_factor = 1 - (3*(self.std_pos+self.std_neg) /
+                             (self.mean_neg-self.mean_pos))
+
+    def control_statistics(self) -> None:
+        """
+        Calculate statistic for control values e.g. two last columns of a plate
+        """
+        pos = self.plate_array[:, -2]
+        neg = self.plate_array[:, -1]
+        self.std_pos = np.std(pos)
+        self.std_neg = np.std(neg)
+        self.mean_pos = np.mean(pos)
+        self.mean_neg = np.mean(neg)
+
+    def find_outliers(self):
+        # TODO
+        ...
+
+    def get_summary_tuple(self) -> list:
+        """
+        Get all features describing a plate in the form of a namedtuple
+
+        :return: namedtuple consisting of plate features
+        """
+        PlateSummary = namedtuple('PlateSummary', ['barcode', 'plate_array',
```

Comment:
suggestion: you can extract the namedtuple definition to the global scope (outside the `Plate` class but within the module) and give the right return type hint (currently it's `list`)

Reply:
Done

Reply:
oops okay, thanks for explanation!

Reply:
just updated the branch, hopefully for the last time